src/errorpage.cc

```diff
@@ -758,6 +758,22 @@ ErrorState::Convert(char token, bool building_deny_info_url, bool allowRecursion
             p = "-";
         break;
 
+    case 'A': {
+        char *tmp = NULL;
+        const bool interceptedAtKnownPort = request ?
```

Comment:
Please do not duplicate this complicated code. I suggest moving this logic, including the `isAnyAddr()` check and the `tcpClient` clause, into a global function. Here is an interface sketch:

```
/// \returns a pointer to known *_port address where the request was accepted
/// \returns nil for nil requests and when the port address info was lost
const Ip::Address *FindListeningPort(const HttpRequest *);
```

And a usage sketch:

```
if (const auto addr = FindListeningPort(request))
    tmp = addr->toStr(ntoabuf, MAX_IPSTRLEN);
```

Both `interceptedAtKnownPort` uses should be updated accordingly.